### PR TITLE
Laksh network topology

### PR DESCRIPTION
**Pull request by lakshbhasin**
This PR addresses a few small things in the component code, and also sets up the network_topology and initializer modules.

### CODE REVIEW

File: network_simulator/network_topology.py [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+            if device.address == device_addr:
+                return device
+
+        return None
+
+    def __get_host_with_addr(self, host_addr):
+        # Host address should be unique among Devices
+        host = self.__get_device_with_addr(host_addr)
+        if host is not None and not isinstance(host, Host):
+            raise ValueError("Tried to get Host with address " + host_addr +
```
Comment (sharonyang) [inline network_simulator/network_topology.py:206]: Do we still need error.py then?

Comment (lakshbhasin) [inline network_simulator/network_topology.py:206]: For the most part, ValueError is fine for indicating invalid values in Python. We can add more specific subclasses but the code will literally be "class CustomException(Exception): pass". I personally don't think error.py is necessary, but I think @yubo56 added it to the design doc so you can ask him.

**Comment by lakshbhasin** [comment]
Note: I have initializer.py code written up. I am waiting for the flow.py code to be merged before I update this PR with that code, since I would like to test the initializer is working before pushing my latest commit. @yubo56 and @dluo11, if you have any additional comments on the flow PR (https://github.com/lakshbhasin/network_simulator/pull/5) please let me know soon.

File: network_simulator/network_topology.py (hunk 1 above)
Comment (yubo56) [inline network_simulator/network_topology.py:206]: I did not add Error.py to the design doc. Sorry for the delay.

**Comment by yubo56** [comment]
This also looks fine by me.

File: network_simulator/network_topology.py (hunk 1 above)
Comment (lakshbhasin) [inline network_simulator/network_topology.py:206]: Okay, I don't think we need error.py for now then. We can add it later if we want to make the errors change names but it's not a big deal.

**Pull request merged**